base/sparse/sparsematrix.jl

```diff
@@ -1666,21 +1666,15 @@ broadcast_zpreserving{Tv,Ti}(f::Function, A_1::Union{Array,BitArray,Number}, A_2
 
 ## Binary arithmetic and boolean operators
 
-for (op, pro) in ((+,   :eltype_plus),
-                  (-,   :eltype_plus),
-                  (min, :promote_eltype),
-                  (max, :promote_eltype),
-                  (&,   :promote_eltype),
-                  (|,   :promote_eltype),
-                  ($,   :promote_eltype))
+for op in (+, -, min, max, &, |, $)
     body = gen_broadcast_body_sparse(op, true)
     OP = Symbol(string(op))
     @eval begin
         function ($OP){Tv1,Ti1,Tv2,Ti2}(A_1::SparseMatrixCSC{Tv1,Ti1}, A_2::SparseMatrixCSC{Tv2,Ti2})
             if size(A_1,1) != size(A_2,1) || size(A_1,2) != size(A_2,2)
                 throw(DimensionMismatch(""))
             end
-            Tv = ($pro)(A_1, A_2)
+            Tv = promote_eltype_op($op, A_1, A_2)
```

Comment:
> Since you have the types as parameters in the function is better to use `promote_op` directly.

Not sure why, but I don't see any harm either, update is coming.